src/main/java/org/breedinginsight/brapps/importer/services/processors/experiment/appendoverwrite/factory/entity/PendingStudy.java

```diff
@@ -0,0 +1,196 @@
+            return;
+        }
+
+        for (U member : members) {
+            BrAPIStudy study = (BrAPIStudy) member;
+
+            // set the DbId for each newly created study
+            String createdStudy_name_no_key = Utilities.removeProgramKeyAndUnknownAdditionalData(study.getStudyName(), importContext.getProgram().getKey());
+            cache.getStudyByNameNoScope().get(createdStudy_name_no_key).getBrAPIObject().setStudyDbId(study.getStudyDbId());
+
+            // Set the study dbId for observation units
+            cache.getObservationUnitByNameNoScope().values()
+                    .stream()
+                    .filter(obsUnit -> obsUnit.getBrAPIObject()
+                            .getStudyName()
+                            .equals(Utilities.removeProgramKeyAndUnknownAdditionalData(study.getStudyName(), importContext.getProgram().getKey())))
+                    .forEach(obsUnit -> {
+                        obsUnit.getBrAPIObject().setStudyDbId(study.getStudyDbId());
+                        obsUnit.getBrAPIObject().setTrialDbId(study.getTrialDbId());
+                    });
+        }
+    }
+
+    /**
+     * Populate the workflow context with objects needed by the workflow.
+     *
+     * @param members List of entities to be initialized
+     */
+    @Override
+    public <U> void initializeWorkflow(List<U> members) {
+        // Check if the input list is of type List<BrAPIStudy>
+        if (experimentUtilities.isInvalidMemberListForClass(members, BrAPIStudy.class)) {
+            return;
+        }
+
+        // Construct the pending studies from the BrAPI studies
+        List<PendingImportObject<BrAPIStudy>> pendingStudies = members.stream()
+                .map(s->(BrAPIStudy) s)
+                .map(pio -> studyService.constructPIOFromBrapiStudy(pio, importContext.getProgram())).collect(Collectors.toList());
+
+        // Construct a hashmap to look up the pending study by study name with the program key removed
+        Map<String, PendingImportObject<BrAPIStudy>> pendingStudyByNameNoScope = pendingStudies.stream()
+                .collect(Collectors.toMap(pio -> Utilities.removeProgramKeyAndUnknownAdditionalData(pio.getBrAPIObject().getStudyName(), importContext.getProgram().getKey()), pio -> pio));
+
+        // Construct a hashmap to look up the pending study by the observation unit ID of a unit stored in the BrAPI service
+        Map<String, PendingImportObject<BrAPIStudy>> pendingStudyByOUId = cache.getPendingObsUnitByOUId().entrySet().stream()
+                .collect(Collectors.toMap(Map.Entry::getKey,
+                        e -> Optional.ofNullable(e.getValue().getBrAPIObject().getStudyName())
+                                .map(studyNameScoped -> Utilities.removeProgramKeyAndUnknownAdditionalData(studyNameScoped, importContext.getProgram().getKey()))
+                                .map(nameNoScope -> pendingStudyByNameNoScope.get(nameNoScope))
```

Comment:
Use method reference
```suggestion
                                .map((pendingStudyByNameNoScope::get)
```